### PR TITLE
Special handling of various field types

### PR DESCRIPTION
**Pull request by jjbennett**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))~~
~~Base axe-core JEST test included for any new LWC (No critical violations)~~
- [x] CRUD/FLS is enforced in Apex
~~Permission sets are updated to account for CRUD/FLS for new object metadata~~
- [x] All modified classes are unit tested (Apex) at 100% coverage
- [x] UX approval or UX not necessary
- [ ] PR contains draft release notes
- [x] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by jjbennett** [comment]
**lurch: attach W-037534

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HoiEQAS" target="_blank">W-037534</a>

**Comment by jjbennett** [comment]
**lurch: attach W-042636

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007hcXAQAY" target="_blank">W-042636</a>

File: force-app/main/default/classes/FieldSetService.cls [hunk 1]
```diff
@@ -8,6 +8,8 @@
  */
 
 public with sharing class FieldSetService {
+    @TestVisible
+    private Set<String> unsupportedFieldTypes = new Set<String>{ 'TIME' };
```
Comment (bethbrains) [inline force-app/main/default/classes/FieldSetService.cls:12]: This makes me very sad.

File: force-app/main/default/classes/FieldSetService.cls [hunk 2]
```diff
@@ -38,10 +39,42 @@ public with sharing class FieldSetService {
...
 
         for (Schema.FieldSetMember field : fieldSet.fields) {
-            fields.add(getFieldForLWC(field));
+            if (
+                unsupportedFieldTypes.contains(field.getType().name()) ||
+                field.getFieldPath().contains('.')
+            ) {
+                // Skipping unsupported field types and fields from related objects.
+                // NOTE: If we ever determine we want to allow reference fields
+                // we can overload this method with a Boolean flag
```
Comment (bethbrains) [inline force-app/main/default/classes/FieldSetService.cls:48]: 👍

File: force-app/main/default/classes/FieldSetService.cls [hunk 3]
```diff
@@ -51,15 +84,38 @@ public with sharing class FieldSetService {
...
-    public Map<String, Object> getFieldForLWC(FieldSetMember field) {
-        String label = field.getLabel();
-
+    @TestVisible
+    private Map<String, Object> getFieldForLWC(DescribeFieldResult field) {
         return new Map<String, Object>{
-            'apiName' => field.getFieldPath(),
-            'label' => label,
+            'apiName' => field.getName(),
+            'label' => getLabel(field),
```
Comment (bethbrains) [inline force-app/main/default/classes/FieldSetService.cls:91]: Thrilled to get this added over here.

File: force-app/main/default/classes/FieldSetService_TEST.cls [hunk 4]
```diff
@@ -7,54 +7,138 @@
...
 
         FieldSetService fieldSetService = new FieldSetService();
-
         Schema.FieldSet fieldSet = Schema.SObjectType.ProgramEngagement__c.FieldSets.CreateProgramEngagement;
 
         List<Map<String, Object>> fieldsResult = fieldSetService.getFieldSetForLWC(
-            'ProgramEngagement__c',
-            'CreateProgramEngagement'
+            String.valueOf(ProgramEngagement__c.SObjectType),
+            fieldSet.getName()
```
Comment (bethbrains) [inline force-app/main/default/classes/FieldSetService_TEST.cls:21]: good improvement.

File: force-app/main/default/classes/FieldSetService_TEST.cls [hunk 5]
```diff
@@ -7,54 +7,138 @@
...
+            'Expected the Compound Field to be returned.'
+        );
+    }
+
+    @IsTest
+    private static void shouldReturnUILabel() {
+        Map<DescribeFieldResult, String> expectedLabelByField = new Map<DescribeFieldResult, String>{
+            SObjectType.Contact.fields.AccountId => 'Account Name',
+            SObjectType.Opportunity.fields.ContactId => 'Full Name',
+            SObjectType.Account.fields.OwnerId => 'Owner'
```
Comment (bethbrains) [inline force-app/main/default/classes/FieldSetService_TEST.cls:133]: should we use the same references for the values here or does that end up being an A=A test?

Comment (jjbennett) [reply force-app/main/default/classes/FieldSetService_TEST.cls:133]: Totally, I can't think of any reason not to.

**Pull request merged**